app/message-table.tsx

```diff
@@ -17,19 +20,65 @@ import {
   TableHeader,
   TableRow,
 } from "@/components/ui/table";
-
-import { useState } from "react";
+import { Checkbox } from "@/components/ui/checkbox";
 
 interface MessageTableProps<TData, TValue> {
   columns: ColumnDef<TData, TValue>[];
   data: TData[];
   defaultExpanded?: boolean;
+  canHideMessages?: boolean;
+  impressionsThreshold?: string;
+}
+
+function MessageTableHead(
+  header: any,
+  hideMessages: boolean,
+  setHideMessages: any,
+  canHideMessages?: boolean,
+  impressionsThreshold?: string,
+) {
+  return (
+    <TableHead className="bg-stone-100 text-slate-400" key={header.id}>
+      {header.isPlaceholder ? null : (
+        <div>
+          {flexRender(header.column.columnDef.header, header.getContext())}
+          {canHideMessages &&
+          header.column.columnDef.header!.toString().includes("Metrics") ? (
+            <div className="flex items-center gap-x-1">
```

Comment:
Overall, even the MessageTabHead function still feels like it has a lot of contextual stuff, and it uses ternary operators (which are hard to read when they're separated in space by a lot of other stuff, but in general, ternary operators are just harder to read than the `if` statements unless their super short).

What do you think about hoisting this `<div>` out into its own functional component as well?

Reply:
Yeah, I think because the `flexRender` logic was there in the original code and part of tanstack tables in general, it might make more sense to hoist the `<div>` part out. The `<div>` part contains most of the checkbox toggle logic anyway! I'll also try to clean up the ternary operation too!